src/backend/wayland/input/keyboard/mmap.rs

```diff
@@ -0,0 +1,91 @@
+//! Temporary implementation of memory mapping, to allow testing keyboard interaction
+use nix::sys::mman::{mmap, munmap, MapFlags, ProtFlags};
+use std::{
+    convert::TryInto,
+    ops::{Deref, DerefMut},
+    os::{raw::c_void, unix::prelude::RawFd},
+    ptr::{self, NonNull},
+    slice,
+};
+pub struct Mmap {
```

Comment:
Could we not use the memmap2 crate?

Reply:
That is the plan, via the rust-x-bindings crate

However, that may require changes in the X11 backend as well, so I'm intending to include that (+ composing) in a follow up PR.

Reply:
I though the X11 backend was going away as well, or is this a kind of 'don't fix what ain't broken' type situation?

Reply:
We're keeping the X11 backend, but we're not improving it

That is, because it mostly works, it's useful as a fallback